src/main/java/com/southernartsoap/controller/ProductController.java

```diff
@@ -0,0 +1,65 @@
+package com.southernartsoap.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.southernartsoap.model.Product;
+import com.southernartsoap.service.ProductService;
+
+public class ProductController {
+	@Autowired
+	private ProductService productService;
+
+	@GetMapping(value = { "/shop" })
+	public String getProducts(@RequestParam(value = "filter", required = false) String filter, Model model) {
+		List<Product> products = new ArrayList<>();
+		if (filter == null) {
```

Comment:
should use "else if" sequence for this.

Reply:
oh nice catch. I was copy+pasting and forgot to add the elses